src/dc_aware_policy.cpp

```diff
@@ -94,11 +94,21 @@ void DCAwarePolicy::on_remove(const Host::Ptr& host) {
 }
 
 void DCAwarePolicy::on_up(const Host::Ptr& host) {
-  on_add(host);
+  const std::string& dc = host->dc();
```

Comment:
I'm pretty sure the driver shares the same instance of hosts throughout the driver and they should be marked UP/DOWN by the control connection and/or session. I think this could be fixed by just removing `on_add()` and `on_remove()` from these methods.